subsys/net/ip/net_tls.c

```diff
@@ -5,14 +5,29 @@
  */
 
 #include "net_tls.h"
+#include "net_private.h"
 
 static int tls_tx(void *ctx, const unsigned char *buf, size_t len )
 {
 	struct net_context *context = ctx;
+	struct net_pkt *pkt;
+	int bytes, ret;
 
-	ARG_UNUSED(context);
+	pkt = net_pkt_get_tx(context, K_FOREVER);
```

Comment:
We need to be very careful with infinite timeout as that can lead to deadlocks. This needs to have a timeout, and we need to bail out properly in this case.

Reply:
Indeed, as this API is advertised to work on "net_context" level, it would need to honor corresponding parameters of its API calls: https://github.com/zephyrproject-rtos/zephyr/pull/7118/commits/c2c424758ae7b76069a746492d846d6e85c76b52#r185754772